utils/format.sh

```diff
@@ -5,12 +5,58 @@ set -euo pipefail
 SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 cd "$SCRIPT_DIR/.."
 
-if ! command -v clang-format; then
-    echo "ERROR: can't find clang-format in your path."
+CLANG_COMMAND=${CLANG_COMMAND:-clang-format}
+
+print_usage() {
+  echo "Missing format option:"
+  echo "  Run: './format.sh fix' if you want to fix format."
+  echo "  Run: './format.sh check' if you want to check format."
+}
+
+fix_format() {
+  find lib tests/unittests/ tools/ include examples \
+    -name \*.h -print0 \
+    -o -name \*.cpp -print0 \
+  | xargs -0 -P8 -n1 $CLANG_COMMAND -i;
+}
+
+check_format() {
+  touch pre.status
+  touch post.status
+
+  git status > pre.status
+  fix_format
+  git status > post.status
+
+  if ! diff pre.status post.status; then
+    echo "ERROR: files need to be formatted"
+    echo "pre.status"
+    cat pre.status
+    echo "post.status"
+    cat post.status
+
```

Comment:
Can we also have git diff printed? It's very annoying to know that something is wrong, but not know what exactly?

Reply:
Yup, good idea!